baseplate/lib/prometheus_metrics.py

```diff
@@ -88,3 +89,81 @@ def get_requests_total_metric(self) -> Counter:
     def get_active_requests_metric(self) -> Gauge:
         """Return the active_requests metrics"""
         return thrift_server_active_requests
+
+
+# http server labels and metrics
+http_server_histogram_labels = [
+    "http_method",
+    "http_endpoint",
+    "http_success",
+]
+http_server_requests_total_labels = [
+    "http_method",
+    "http_endpoint",
+    "http_success",
+    "http_response_code",
+]
+http_server_active_requests_labels = [
+    "http_method",
+    "http_endpoint",
+]
+
+http_server_latency_seconds = Histogram(
+    "http_server_latency_seconds",
+    "Time spent processing requests",
+    http_server_histogram_labels,
+    buckets=default_buckets,
+)
+
+http_server_requests_total = Counter(
+    "http_server_requests_total",
+    "Total number of request handled",
+    http_server_requests_total_labels,
+)
+http_server_active_requests = Gauge(
+    "http_server_active_requests",
+    "Current requests in flight",
+    http_server_active_requests_labels,
+)
+
+
+class PrometheusHTTPServerMetrics:
+    def __init__(self) -> None:
+        pass
+
+    def latency_seconds_metric(self, tags: Dict[str, str]) -> Any:
+        return http_server_latency_seconds.labels(
+            http_method=tags.get("http.method", ""),
+            http_endpoint=tags.get("http.route", ""),
+            http_success=getHTTPSuccessLabel(int(tags.get("http.status_code", "0"))),
+        )
+
+    def requests_total_metric(self, tags: Dict[str, str]) -> Any:
+        return http_server_requests_total.labels(
+            http_method=tags.get("http.method", ""),
+            http_endpoint=tags.get("http.route", ""),
+            http_success=getHTTPSuccessLabel(int(tags.get("http.status_code", "0"))),
+            http_response_code=tags.get("http.status_code", ""),
+        )
+
+    def active_requests_metric(self, tags: Dict[str, str]) -> Any:
+        return http_server_active_requests.labels(
+            http_method=tags.get("http.method", ""),
+            http_endpoint=tags.get("http.route", ""),
+        )
+
+    def get_latency_seconds_metric(self) -> Any:
+        return http_server_latency_seconds
+
+    def get_requests_total_metric(self) -> Any:
+        return http_server_requests_total
+
+    def get_active_requests_metric(self) -> Any:
+        return http_server_active_requests
+
+
+def getHTTPSuccessLabel(httpStatusCode: int) -> str:
+    """
+    The HTTP success label is "true" if the status code is 2xx or 3xx, "false" otherwise.
+    """
+    return str(httpStatusCode >= 200 < 400).lower()
```

Comment:
Could you add a unit test for this too please?

Reply:
Thanks, that was wrong :sweat_smile: Changed it without thinking because the linter didn't like it being two expressions. Added tests.